tests/e2e/collectives/collectives_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import unittest
+import iree.compiler
+import argparse
+import sys
+import iree.runtime
+from iree.runtime.array_interop import DeviceArray
+import os
+from numpy.typing import ArrayLike
```

Comment:
https://github.com/openxla/iree/actions/runs/5626159982/job/15295889995?pr=14418#step:7:232

```
ModuleNotFoundError: No module named 'numpy.typing'
```

Is that resolved by `mpi4py==3.1.4`? We might need to rebuild the Docker images to pull in any new python package deps.

Reply:
I added `numpy` to the run requirements of the test.